shell/platform/android/io/flutter/embedding/android/FlutterFragment.java

```diff
@@ -76,238 +72,268 @@
+   *
+   * Subclasses can alter the {@link FlutterEngine} after creation by overriding
+   * {@link #onFlutterEngineCreated(FlutterEngine)}.
    */
-  public void onBackPressed() {
-    Log.d(TAG, "onBackPressed()");
-    popRoute();
+  private void createFlutterEngine() {
+    // Create a FlutterEngine to back our FlutterView.
+    flutterEngine = onCreateFlutterEngine(getActivity());
+
+    // Allow subclasses to customize FlutterEngine as desired.
+    onFlutterEngineCreated(flutterEngine);
   }
 
   /**
-   * The result of a permission request has been received.
-   *
-   * See {@link Activity#onRequestPermissionsResult(int, String[], int[])}
+   * Hook for subclasses to customize the creation of the {@code FlutterEngine}.
    *
-   * @param requestCode identifier passed with the initial permission request
-   * @param permissions permissions that were requested
-   * @param grantResults permission grants or denials
+   * By default, this method returns a standard {@link FlutterEngine} without any modification.
    */
-  public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
-    flutterEngine.getPluginRegistry().onRequestPermissionsResult(requestCode, permissions, grantResults);
+  @NonNull
+  protected FlutterEngine onCreateFlutterEngine(@NonNull Context context) {
+    Log.d(TAG, "onCreateFlutterEngine()");
+    return new FlutterEngine(context);
   }
 
   /**
-   * A new Intent was received by the {@link Activity} that currently owns this {@link Fragment}.
-   *
-   * See {@link Activity#onNewIntent(Intent)}
-   *
-   * @param intent new Intent
+   * Hook for subclasses to customize the {@link FlutterEngine} owned by this {@link FlutterFragment}
+   * after the {@link FlutterEngine} has been instantiated.
    */
-  public void onNewIntent(@NonNull Intent intent) {
-    flutterEngine.getPluginRegistry().onNewIntent(intent);
-  }
-
-  @Override
-  public void onActivityResult(int requestCode, int resultCode, Intent data) {
-    flutterEngine.getPluginRegistry().onActivityResult(requestCode, resultCode, data);
+  protected void onFlutterEngineCreated(@NonNull FlutterEngine flutterEngine) {
+    // no-op
   }
 
   /**
-   * The {@link Activity} that owns this {@link Fragment} is about to go to the background
-   * as the result of a user's choice/action, i.e., not as the result of an OS decision.
+   * Creates a {@link PlatformPlugin} that receives requests from the app running in the
+   * {@link FlutterEngine} and carries out the corresponding Android platform behavior.
    *
-   * See {@link Activity#onUserLeaveHint()}
+   * The created plugin is connected to the {@link FlutterEngine} via its
+   * {@link io.flutter.embedding.engine.systemchannels.PlatformChannel}.
+   *
+   * Some examples of behavior carried about by the platform plugin include:
+   *  - clipboard
+   *  - haptic feedback
+   *  - playing system sounds
+   *  - etc.
```

Comment:
Use Javadoc list format `<li>` to ensure proper formatting in output.

Reply:
done